menstoreproject/catalog/views.py

```diff
@@ -12,3 +12,22 @@ def index(request):
 class ViewProduct(DetailView):
     model = Product
     context_object_name = 'product_item'
+
+
+class ProductList(ListView):
+    model = Product
+    template_name = 'catalog/product_list.html'
+    context_object_name = 'products'
+
+    def get_queryset(self):
+        return Product.objects.select_related('category')
+
+
+class ProductByCategory(ListView):
+    model = Product
+    template_name = 'catalog/product_list.html'
+    context_object_name = 'products'
+
+    def get_queryset(self):
```

Comment:
my suggestion to return this queryset if category__slug is provided, else return just Product.objects.all()